tests/unit/test_client.py

```diff
@@ -50,16 +51,18 @@ def blob_properties(blob_length, blob_etag):
 
 
 @pytest.fixture
-def mock_generated_sdk_storage_client():
+def mock_generated_sdk_storage_client(blob_length):
     mock_generated_sdk_client = mock.Mock(AzureBlobStorage)
     mock_generated_sdk_client.blob = mock.Mock(BlobOperations)
     return mock_generated_sdk_client
 
 
 @pytest.fixture
-def mock_sdk_blob_client(mock_generated_sdk_storage_client):
+def mock_sdk_blob_client(mock_generated_sdk_storage_client, blob_length):
     mock_sdk_client = mock.Mock(BlobClient)
     mock_sdk_client._client = mock_generated_sdk_storage_client
+    mock_sdk_client._client._blob_properties = mock.Mock()
+    mock_sdk_client._client._blob_properties.size = blob_length
```

Comment:
I'm a little hesitant to be accessing even more internal properties as it generally will make it harder to do refactorings in the future. Could we just set the `return_value` for the `get_blob_properties` method? I think that could make sense especially since most tests are setting the return value for that method.